src/zoo/owl_zoo_cmd.ml

```diff
@@ -28,47 +29,54 @@ let preprocess script =
   tmp_script
 
 
-let remove_gist gist =
-  Owl_log.debug "owl_zoo: %s removed" gist;
-  let dir = Sys.getenv "HOME" ^ "/.owl/zoo/" ^ gist in
+(* TODO: use single gid to express "removing all versions of this gist" *)
+let remove_gist ?(all=false) gist =
+  let gid, vid, _ = Owl_zoo_dir.parse_gist_string gist in
+  let vid = if all then vid else "" in
+  let dir = dir ^ gid ^ "/" ^ vid in
   let cmd = Printf.sprintf "rm -rf %s" dir in
-  Sys.command cmd |> ignore
+  let ret = Sys.command cmd in
+  if ret = 0 then (
+    Owl_zoo_log.remove_log gid vid;
+    Owl_log.debug "owl_zoo: %s removed" gid
+  )
+  else Owl_log.debug "owl_zoo: Error remvoing gist %s" gid
 
 
 let upload_gist gist =
```

Comment:
Does this function support updating one specific version of a gist? This indicates `upload` understands the naming scheme in the same way as `download`, right?

Reply:
In `upload`, though the full gist name is passed in but only the gist id is used.  It does not support updating a specific version.

Reply:
Sorry, my bad. Updating one specific version does not really make sense, every time we upload a gist, we generate a new version. The reason is I saw you called `parse_gist_string` ... then why call this function if you only use gist id.

I think we have the same situation here as for `remove` ... `upload` only needs to understand gist id rather than the naming scheme ... so calling parsing function is a bit pointless ... is it, unless you do extra check there?

Reply:
Yes, it's similar to `remove`: I assumed every command takes the same full gist name, but what this function needs is only `gid`.